apiml-common/src/main/java/org/zowe/apiml/product/eureka/client/ApimlPeerEurekaNode.java

```diff
@@ -372,11 +373,39 @@ public static class ReplicationTaskProcessor implements TaskProcessor<Replicatio
 
         private static final Pattern READ_TIME_OUT_PATTERN = Pattern.compile(".*read.*time.*out.*");
 
+        private final NetworkIssueCounter networkIssueCounter = new NetworkIssueCounter();
+
         public ReplicationTaskProcessor(String peerId, HttpReplicationClient replicationClient) {
             this.replicationClient = replicationClient;
             this.peerId = peerId;
         }
 
+        static class NetworkIssueCounter {
+
+            static final int MAX_RETRIES = 10;
```

Comment:
Why 10? Should it be configurable?

Reply:
Oh no :-D I was expecting this comment :-D I would like to since I also think it is a good idea. But it is not a spring class so I can't easily use @Value. I would have to read this value elsewhere and then use the constructor. And I didn't want to change too many Eureka classes since we have to maintain them with each Eureka update. But feel free to do it if you have an idea that will cause not many changes.

Reply:
Why 10? Just a wild guess that should cover a small network issue but stops repeating it when it is not connected for a longer time.

Reply:
I think it should not be much change, we can add the configuration value in EurekaConfig, because `RefreshablePeerEurekaNodes` is a spring bean, and we can add a property for the constructor in `ApimlPeerEurekaNode`, if you prefer I can try it out

Reply:
Not saying that 10 is a wrong number, just wondering if it's based on some evidence, after release could be late to realise there are valid cases where it could take more than 10 but still recover

Reply:
You can try it out. Just don't do too many changes. I'd like to avoid merging hell when the original Eureka classes are updated. We are just copy/pasting them.

Reply:
[diff.patch](https://github.com/zowe/api-layer/files/10567436/diff.patch)
This is the change I'm thinking about (without the changes to the test classes), I don't know if this is too much change, but it doesn't seem like so to me.

Reply:
This is exactly what I was considering and decided not to do. :-) But push it. It can be useful in the future.

Reply:
I pushed the changes